### PR TITLE
Move initial default redirect to server-side method

### PR DESCRIPTION
**Pull request by mattcasey**
### <samp>🤖 Generated by Copilot at 6485626</samp>

This pull request refactors the login page and the spaces API to use server-side rendering and common functions. It improves the performance, code quality, and user experience of the app. It also fixes a bug in the `joinDate` property of the `getOrgs` function. It modifies the files `components/login/LoginPage.tsx`, `pages/api/spaces/index.ts`, `lib/profile/getOrgs.ts`, `lib/spaces/getSpacesOfUser.ts`, and `pages/index.tsx`.

### CODE REVIEW

File: components/login/LoginPage.tsx [hunk 1]
```diff
@@ -23,34 +19,11 @@ export function LoginPageView() {
   const router = useRouter();
   const [, setTitleState] = usePageTitle();
   const { user, isLoaded } = useUser();
-  const { spaces, isLoaded: isSpacesLoaded } = useSpaces();
   const discordCookie = getCookie(AUTH_CODE_COOKIE);
   const { onClick: openSettingsModal, open: isSettingsDialogOpen } = useSettingsDialog();
-  const [showLogin, setShowLogin] = useState(false); // capture isLoaded state to prevent render on route change
   const isLogInWithDiscord = Boolean(discordCookie);
-  const isDataLoaded = triedEager && isSpacesLoaded && isLoaded;
+  const isDataLoaded = triedEager && isLoaded;
```
Comment (motechFR) [inline components/login/LoginPage.tsx:25]: Style suggestion; Rename isLoaded to isUserLoaded

File: hooks/useUser.tsx [hunk 2]
```diff
@@ -31,8 +30,7 @@ export function UserProvider({ children }: { children: ReactNode }) {
 
   async function logoutUser() {
     await charmClient.logout();
-    setUser(null);
-    clearSWRCache();
+    window.location.href = window.location.origin;
```
Comment (motechFR) [inline hooks/useUser.tsx:33]: Nice clean reload :)

File: lib/session/__tests__/getDefaultPage.spec.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+describe('getDefaultPage()', () => {
+  it('should send user to create a space', () => {
+    const url = getDefaultPage({ spaces: [] });
+    expect(url).toEqual('/createSpace');
+  });
+
+  it('should send user to redirect url', () => {
+    const returnUrl = '/charmverse/getting-started';
+    const url = getDefaultPage({ returnUrl, spaces: [] });
```
Comment (motechFR) [inline lib/session/__tests__/getDefaultPage.spec.ts:11]: If the user has empty spaces (implying they are not a member), wouldn't we want to send them to /join for that space?

Comment (mattcasey) [reply lib/session/__tests__/getDefaultPage.spec.ts:11]: That's what would happen once u send them to that page

**Pull request merged**